plugins/rule_engines/irods_rule_language/src/configuration.cpp

```diff
@@ -594,6 +636,32 @@ int readRuleStructAndRuleSetFromFile( const char *ruleBaseName, const char *rule
     freeRErrorContent( &errmsgBuf );
     return res;
 }
+int readRuleStructAndRuleSetFromBuffer(const char* ruleBaseName, char* ruleBase)
+{
+    int errloc;
+    rError_t errmsgBuf;
+    errmsgBuf.errMsg = NULL;
+    errmsgBuf.len = 0;
+
+    char* buf = (char*) malloc(ERR_MSG_LEN * 1024 * sizeof(char));
+    int res = 0;
+    if ((res = readRuleSetFromBuffer(ruleBaseName,
```

Comment:
Let's move the assignment out of the if-stmt and let the if-stmt handle the condition. That will reduce complexity for other readers.